core/arch/instr.h

```diff
@@ -318,6 +318,13 @@ typedef struct _dr_instr_label_data_t {
     ptr_uint_t data[4]; /**< Generic fields for storing user-controlled data */
 } dr_instr_label_data_t;
 
+/**
+ * Label instruction callback function. Set by instr_set_label_callback() and
+ * called when the label is freed. \p instr is typically an instruction related
+ * to the label which can be freed at the same time as the label.
```

Comment:
The 2nd sentence is incorrect: \p instr is the label instruction itself (and the caller extracts the auxiliary data to be freed from it)

Reply:
Oops, good spot! Misleading for users! Fixed.